homeassistant/components/evohome.py

```diff
@@ -0,0 +1,606 @@
+"""Support for Honeywell evohome (EMEA/EU-based systems only).
+
+Support for a temperature control system (TCS, controller) with 0+ heating
+zones (e.g. TRVs, relays) and, optionally, a DHW controller.
+
+A minimal configuration.yaml is as as below:
+
+evohome:
+  username: !secret evohome_username
+  password: !secret evohome_password
+
+# This config parameter is presented with its default value:
+# location_idx: 0        # if you have more than 1 location, use this
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/evohome/
+"""
+
+# Glossary
+# TCS - temperature control system (a.k.a. Controller, Master), which can
+# have up to 13 Slaves:
+#   0-12 Heating zones (a.k.a. Zone), and
+#   0-1 DHW controller, (a.k.a. Boiler)
+
+import logging
+from datetime import datetime, timedelta
+import requests
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_AWAY_MODE,
+)
+
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+
+    EVENT_HOMEASSISTANT_START,
+    PRECISION_TENTHS,
+
+    TEMP_CELSIUS,
+
+    HTTP_BAD_REQUEST,
+    HTTP_TOO_MANY_REQUESTS,
+)
+
+from homeassistant.core import callback
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+REQUIREMENTS = ['evohomeclient==0.2.7']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Usually, only the controller does client api I/O during update() to pull
+# current state - the exception is when zones pull their own schedules.
+# However, any entity can call methods that will (eventually) change state.
+PARALLEL_UPDATES = 0
```

Comment:
This only works for platforms.

Reply:
I can take it out but I will have to put it back in in an upcoming PR.

When fully implemented, the Controller will pull state during its `update()`, and the other entities will use that data during their `update()` without having to use the client library.

That's why the comment what is does...

Reply:
Moved to `./climate/evohome.py`.